wrapper/src/main/java/software/amazon/jdbc/plugin/readwritesplitting/ReadWriteSplittingPlugin.java

```diff
@@ -165,6 +171,51 @@ private boolean isReader(final @NonNull HostSpec hostSpec) {
     return HostRole.READER.equals(hostSpec.getRole());
   }
 
+  void pickNewReaderConnection() {
+    final List<HostSpec> hosts = this.pluginService.getHosts();
+    if (hosts.size() <= 2) {
+      LOGGER.finest(() -> Messages.get("ReadWriteSplittingPlugin.driverWillNotSwitchToNewReader"));
+      return;
+    }
+
+    final ArrayDeque<HostSpec> readerHosts = getRandomReaderHosts();
+    while (!readerHosts.isEmpty()) {
+      final HostSpec host  = readerHosts.poll();
+      try {
+        getNewReaderConnection(host);
+        final Connection currentConnection = this.pluginService.getCurrentConnection();
+        switchCurrentConnectionTo(currentConnection, this.readerHostSpec);
+
+        LOGGER.finest(
+            () -> Messages.get(
+                "ReadWriteSplittingPlugin.successfullyConnectedToReader",
+                new Object[] {
+                    host.getUrl()}));
+        return;
+      } catch (SQLException e) {
+        LOGGER.config(
+            () -> Messages.get(
+                "ReadWriteSplittingPlugin.failedToConnectToReader",
+                new Object[] {
+                    host.getUrl()}));
+      }
+    }
+    // If we get here we failed to connect to a new reader. In this case we will stick with the current one
+  }
+
+  private ArrayDeque<HostSpec> getRandomReaderHosts() {
+    final List<HostSpec> hosts = this.pluginService.getHosts();
+    final List<HostSpec> readerHosts = new ArrayList<>();
+    for (HostSpec host : hosts) {
+      if (HostRole.READER.equals(host.getRole())
+          && !this.pluginService.getCurrentHostSpec().getUrl().equals(host.getUrl())) {
+        readerHosts.add(host);
+      }
+    }
+    Collections.shuffle(hosts);
+    return new ArrayDeque<>(hosts);
```

Comment:
Are these supposed to be:
```suggestion
    Collections.shuffle(readerHosts);
    return new ArrayDeque<>(readerHosts);
```